docs/custom_regions.qmd

```diff
@@ -0,0 +1,142 @@
+Most custom regions will require some changes to these default settings.
+
+The settings used to run a given region `regnm` on DPS are defined in a `.env` file stored in the `Settings.PREPROCESSED_DIR/${regnm}` folder. If there is no such file here, the default settings are used. 
+
+::: callout-note
+`Settings.PREPROCESSED_DIR` renders out to be `s3://maap-ops-workspace/shared/gsfc_landslides/FEDSpreprocessed` in our current production setup.
+:::
+
+A couple points will describe how this all works: 
+
+1. We use the package [pydantic-settings](https://github.com/pydantic/pydantic-settings) in the fireatlas code to manage 
+our settings. If you look at the class declaration in `FireConsts.py` you'll see this piece of config below. This tells us
+a few different things:
+
+    ```bash
+   class Settings(BaseSettings):
+    # read in all env vars prefixed with `FEDS_` they can be in a .env file
+    model_config = {
+        "env_file": ".env",
+        "extra": "ignore",
+        "env_prefix": "FEDS_",
+    }
+    ```
+
+    * if there is a `.env` file available read it and use the key/values there to override any defaults listed in this class
+    * if there are variables in the `.env` file that are not declared in this class do not use them
+    * everything in the `.env` that will act as an override is prefixed with `FEDS_` 
+    (so to override `LOCAL_PATH` in the `.env` you would declare `FEDS_LOCAL_PATH=<value>`)
+
+2. All DPS jobs are kicked off via the `/maap_runtime/run_dps_cli.sh` script. In that file there's a single function and line
+that attempts (and gracefully exists) to bring over the `.env` file for a DPS run:
+
+    ```bash
+    # TODO: this will have to be changed to be passed dynamically once we want to use other s3 buckets
+    copy_s3_object "s3://maap-ops-workspace/shared/gsfc_landslides/FEDSpreprocessed/${regnm}/.env" ../fireatlas/.env
+    ```
+   
+Any imports of the `fireatlas` package should now have these overrides included in calls to `python FireRunDaskCoordinator.py`. 
+
+## Defining a New Region 
+
+In the future, this should be a more rigid and exclusively additive workflow. For now, there is only the requirement to pick a unique, descriptive region name **that does not already exist** in `FEDSpreprocessed`. The example below will use the newly created "RussiaEast" region to explain how this is done. 
+
+1. Pick a unique, descriptive region name (e.g. "Russia East")
+
+2. Create a .env file defining the settings you want to use for your region. For example: 
+
+    ```bash
+    FEDS_FTYP_OPT="global"
+    FEDS_CONT_OPT="global"
+    FEDS_EPSG_CODE=6933
+    ```
+::: callout-tip
+Working with hidden files (prefixed by a `.`) is difficult in the MAAP ADE, because the GUI file explorer has no way to show them. Use the terminal instead. Another thing you can do is create a file called `env`, edit it with the GUI as needed, then rename it to `.env` when you are done. 
```

Comment:
very helpful! and just to clarify `env` does **not** need an extension, right?

Reply:
yup, that is right